external/kafka/src/test/scala/org/apache/spark/streaming/kafka/DirectKafkaStreamSuite.scala

```diff
@@ -381,3 +447,20 @@ object DirectKafkaStreamSuite {
     }
   }
 }
+
+private[streaming] class ConstantEstimator(rates: Double*) extends RateEstimator {
```

Comment:
Aah, the kafka project may not have spark tests as a test dependency. Dont worry about it then. Though I suggest making the test more deterministic like I am doing in my PR. You could copy the modified ConstantEstimator that I am using and use that to make the tests more deterministic. I can take care of deduping it later.